dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/Protections/LossOfSynchronismProtection.mo

```diff
@@ -0,0 +1,49 @@
+within Dynawo.Electrical.Controls.Machines.Protections;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model LossOfSynchronismProtection "Loss of synchronism protection for generators"
+  /* Trips when the internal angle (variable thetaInternal) of a generator is negative and lower than -abs(ThetaMin). The protection never disarms because there is a risk to "loop around" (thetaInternal is always between -pi and pi).*/
+  import Modelica.Constants;
+  import Modelica.Blocks.Interfaces.RealInput;
+  import Dynawo.NonElectrical.Logs.Timeline;
+  import Dynawo.NonElectrical.Logs.TimelineKeys;
+
+  parameter Types.Angle ThetaMin "Angle threshold above which the automaton is activated in rad. It is recommended to use a value that is not to close to (-)pi as the simulator migth skip it.";
```

Comment:
It is recommended to use a value not too close to pi or -pi as the simulator might skip it.